flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/ImplicitConversionEqualsFunctionITCase.java

```diff
@@ -0,0 +1,227 @@
+import org.apache.flink.table.types.DataType;
+
+import org.junit.runners.Parameterized;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.apache.flink.table.api.DataTypes.BIGINT;
+import static org.apache.flink.table.api.DataTypes.BOOLEAN;
+import static org.apache.flink.table.api.DataTypes.DATE;
+import static org.apache.flink.table.api.DataTypes.DECIMAL;
+import static org.apache.flink.table.api.DataTypes.DOUBLE;
+import static org.apache.flink.table.api.DataTypes.FLOAT;
+import static org.apache.flink.table.api.DataTypes.INT;
+import static org.apache.flink.table.api.DataTypes.SMALLINT;
+import static org.apache.flink.table.api.DataTypes.STRING;
+import static org.apache.flink.table.api.DataTypes.TIME;
+import static org.apache.flink.table.api.DataTypes.TIMESTAMP;
+import static org.apache.flink.table.api.DataTypes.TINYINT;
+
+/** Tests for type conversions in '='. */
+public class ImplicitConversionEqualsFunctionITCase extends BuiltInFunctionTestBase {
+
+    // numeric data
+    private static final byte TINY_INT_DATA = (byte) 1;
+    private static final short SMALL_INT_DATA = (short) 1;
+    private static final int INT_DATA = 1;
+    private static final long BIG_INT_DATA = 1L;
+    private static final float FLOAT_DATA = 1.0f;
+    private static final double DOUBLE_DATA = 1.0d;
+    private static final BigDecimal DECIMAL_DATA = new BigDecimal(1);
+
+    // time data
+    private static final String DATE_DATA = "2001-01-01";
+    private static final String TIME_DATA = "00:00:00";
+    private static final String TIMESTAMP_DATA = ("2001-01-01 00:00:00");
+
+    // string data
+    private static final String STRING_DATA_EQUALS_NUMERIC = "1";
+    private static final String STRING_DATA_EQUALS_DATE = "2001-01-01";
+    private static final String STRING_DATA_EQUALS_TIME = "00:00:00";
+    private static final String STRING_DATA_EQUALS_TIMESTAMP = "2001-01-01 00:00:00";
+
+    @Parameterized.Parameters(name = "{index}: {0}")
+    public static List<TestSpec> testData() {
+        final List<TestSpec> specs = new ArrayList<>();
+        specs.addAll(implicitConversionBetweenNumeric());
+        specs.addAll(implicitConversionBetweenTimeAndString());
+        specs.addAll(unsupportedImplicitConversionBetweenNumericAndString());
+        return specs;
+    }
+
+    private static List<TestSpec> implicitConversionBetweenNumeric() {
+        return Arrays.asList(
+                TypeConversionTestBuilder.left(TINYINT(), TINY_INT_DATA)
+                        .right(TINYINT(), TINY_INT_DATA)
+                        .right(SMALLINT(), SMALL_INT_DATA)
+                        .right(INT(), INT_DATA)
+                        .right(BIGINT(), BIG_INT_DATA)
+                        .right(FLOAT(), FLOAT_DATA)
+                        .right(DOUBLE(), DOUBLE_DATA)
+                        .right(DECIMAL(1, 0), DECIMAL_DATA)
```

Comment:
It seems once anyone wants to change decimal value he/she should remember to change this and other similar lines...

Reply:
Yes, because these test cases are about implicit type conversions between different types with function '=' and we only test the case in which the result is TRUE. To ensure the result is TRUE, these values should be same. The reason why we only test TRUE is aimed to verify fixing the bug that implicit type conversion between numeric and (var)char in '=' was always caused FALSE.So in order to test the correctness, it's meaningless to test the result FALSE again.